tool/build.dart

```diff
@@ -200,19 +231,26 @@ _main(List<String> argv) async {
       }
     }
 
-    var cr = await exec("git", args: [
+    var gitArgs = <String>[
       "clone",
-      "--depth=1",
-      repo,
-      "."
-    ], writeToBuffer: true);
+      "--depth=1"
+    ];
+
+    if (branch != "HEAD") {
+      gitArgs.add("--branch=${branch}");
+    }
+
+    gitArgs.add(repo);
+    gitArgs.add(".");
+
+    var cr = await exec("git", args: gitArgs, writeToBuffer: true);
     if (cr.exitCode != 0) {
       await fail("DSLink ${name}: Failed to clone repository.\n${cr.output}");
     }
 
     var rpo = await exec("git", args: [
       "rev-parse",
-      "HEAD"
+      branch
```

Comment:
We want to get the current commit for the requested branch on the local repository.